File: src/main/java/gg/mineral/practice/commands/stats/EloCommand.java
package gg.mineral.practice.commands.stats;

import gg.mineral.practice.commands.PlayerCommand;
import gg.mineral.practice.inventory.menus.EloMenu;
import gg.mineral.practice.managers.ProfileManager;
import gg.mineral.practice.managers.QueuetypeManager;
import gg.mineral.practice.queue.Queuetype;
import lombok.val;

public class EloCommand extends PlayerCommand {

	public EloCommand() {
		super("elo");
	}

	@Override
	public void execute(org.bukkit.entity.Player pl, String[] args) {
		val profile = ProfileManager.getOrCreateProfile(pl);

		QueuetypeManager.getQueuetypes().values().stream().filter(Queuetype::isRanked).findAny()
				.ifPresent(queuetype -> {
					if (args.length == 0) {
						profile.openMenu(new EloMenu(profile, queuetype));
						return;
					}

					val eloProfile = ProfileManager.getProfileData(args[0], null);
					profile.openMenu(new EloMenu(eloProfile, queuetype));
				});
	}
}

File: src/main/kotlin/gg/mineral/practice/commands/stats/EloCommand.kt
package gg.mineral.practice.commands.stats

import dev.rollczi.litecommands.annotations.argument.Arg
import dev.rollczi.litecommands.annotations.command.Command
import dev.rollczi.litecommands.annotations.context.Context
import dev.rollczi.litecommands.annotations.execute.Execute
import gg.mineral.practice.entity.Profile
import gg.mineral.practice.entity.ProfileData
import gg.mineral.practice.inventory.menus.EloMenu
import gg.mineral.practice.managers.QueuetypeManager.queuetypes
import java.util.*

@Command(name = "elo")
class EloCommand {
    @Execute
    fun execute(@Context profile: Profile, @Arg profileData: Optional<ProfileData>) {
        // TODO: Support multiple ranked queuetypes
        queuetypes.values.first { it?.ranked == true }.let { queuetype ->
            profileData.ifPresentOrElse({ eloProfile ->
                queuetype?.let { EloMenu(eloProfile, it) }?.let { profile.openMenu(it) }
            }, {
                queuetype?.let { EloMenu(profile, it) }?.let { profile.openMenu(it) }
            })
        }
    }
}
